tests/scene/test_occluder_3d.h

```diff
@@ -0,0 +1,250 @@
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#pragma once
+
+#include "core/math/vector3.h"
+#include "scene/3d/occluder_instance_3d.h"
+
+#include "tests/test_macros.h"
+/*  At least on my system (windows 11 laptop, compiling using msvc)
+	when comparing two packedvector3array's, -0.0!=0.0
+	and the sphere occluder vertices for some reason have random signed zeroes sprinkled about (bug?)
+	so i have to use this function to compare the arrays using is_equal_approx  */
+bool is_packedVector3Array_equal(PackedVector3Array check_one, PackedVector3Array check_two) {
+	MESSAGE("equal check started");
+	for (int y = 0; y < check_one.size(); y++) {
+		MESSAGE("size check");
+		if (check_two.size() < check_one.size() || check_two.size() > check_one.size()) {
+			return false;
+		};
+		MESSAGE("item check");
+		if (not check_one[y].is_equal_approx(check_two[y])) {
+			return false;
+		};
+		MESSAGE("item check done");
+	};
+	return true;
+}
```

Comment:
The prints are quite spammy when running tests, so I'd remove them:

```suggestion
bool array_is_equal_approx(PackedVector3Array check_one, PackedVector3Array check_two) {
	for (int y = 0; y < check_one.size(); y++) {
		if (check_two.size() < check_one.size() || check_two.size() > check_one.size()) {
			return false;
		};
		if (not check_one[y].is_equal_approx(check_two[y])) {
			return false;
		};
	};
	return true;
}
```

Also, I suggest renaming the function to `array_is_equal_approx()` to match the style guide.